src/tests/functional/inference_engine/transformations/smart_reshape/lstm_states_broadcast.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <openvino/core/model.hpp>
+#include <openvino/opsets/opset9.hpp>
+
+#include "common_test_utils/ngraph_test_utils.hpp"
+
+template<class T>
+std::shared_ptr<ov::opset9::Constant> create_constant(const std::vector<T>& data, const ov::element::Type_t et = ov::element::i64, bool scalar = false) {
+    ov::Shape shape = scalar ? ov::Shape{} : ov::Shape{data.size()};
+    return ov::opset9::Constant::create(et, shape, data);
+}
+
+std::shared_ptr<ov::opset9::Constant> create_zero_constant(const ov::element::Type_t et, ov::PartialShape shape) {
+    return ov::opset9::Constant::create(et, shape.to_shape(), {0});
+}
+
+struct LSTMStatesAttributes {
+    ov::element::Type_t data_et;
+    ov::Dimension data_batch_size, new_batch_size;
+    ov::Dimension input_size, hidden_size;
+};
+
+class LSTMStatesBroadcastTest
+        : public testing::WithParamInterface<LSTMStatesAttributes>, public CommonTestUtils::TestsCommon {
+};
+
+TEST_P(LSTMStatesBroadcastTest, BareLSTM) {
+    auto p = GetParam();
+
+    std::shared_ptr<ov::Model> model(nullptr);
+    {
+        auto parameter = std::make_shared<ov::opset9::Parameter>(p.data_et, ov::PartialShape{p.data_batch_size, p.input_size});
+        auto initial_hidden_state = create_zero_constant(p.data_et, {1, p.hidden_size});
+        auto initial_cell_state = create_zero_constant(p.data_et, {1, p.hidden_size});
+        auto W = create_zero_constant(p.data_et, {p.hidden_size * 4, p.input_size});
```

Comment:
Why mul by 4? Make this number as `consexpr` with more meaningful name?

Reply:
I've filled the data with respect to specification of LSTMCell operation. It is documented there